cni/ipam/plugin/main.go

```diff
@@ -42,11 +42,19 @@ func main() {
 
 	if err := ipamPlugin.Plugin.InitializeKeyValueStore(&config); err != nil {
 		fmt.Printf("Failed to initialize key-value store of ipam plugin, err:%v.\n", err)
+
+		if isSafe, err := ipamPlugin.Plugin.IsSafeToRemoveLock(ipamPlugin.Plugin.Name); isSafe {
+			log.Printf("[IPAM] Removing lock file as process holding lock exited. Error:%v", err)
```

Comment:
You are not logging the err when isSafe is false - which is where we get non-nil err.
You are always logging nil error.